store/sqlstore/thread_store.go

```diff
@@ -483,33 +483,31 @@ func (s *SqlThreadStore) GetThreadForUser(teamId string, threadMembership *model
 			sq.Eq{"Posts.DeleteAt": 0},
 		})
 
-	fetchConditions := sq.And{
-		sq.Or{
-			sq.Eq{"Threads.ThreadTeamId": teamId},
-			// Include DMs & GMs
-			sq.Eq{"Threads.ThreadTeamId": ""},
-		},
-		sq.Eq{"Threads.PostId": threadMembership.PostId},
+	threadQuery := s.threadsAndPostsSelectQuery.
+		Where(sq.Eq{"Threads.PostId": threadMembership.PostId})
+	var thread JoinedThread
+	err := s.GetReplicaX().GetBuilder(&thread, threadQuery)
```

Comment:
This effectively changes things from a single query (with a complicated fetch) to dual queries -- is it possible to retain the original semantics of just one query, and simply drop the `ThreadTeamId` checks?